File: front/src/app/Services/Activity/activity.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environemnt } from '../../../environment/Environment';
import { Activity, CreateActivity, CreateProposal, Proposal } from '../../interfaces/Activity';
import { Observable } from 'rxjs';
import { ProposalState } from '../../interfaces/Enums';

@Injectable({
  providedIn: 'root'
})
export class ActivityService {
  baseUrl = environemnt.baseUrl + '/activities'
  constructor(private http: HttpClient) { }

  createActivity(data: CreateActivity): Observable<Activity> {
    return this.http.post<Activity>(this.baseUrl, data);
  }

  getActivitiesByCongressId(congressId: number): Observable<Activity[]> {
    return this.http.get<Activity[]>(`${this.baseUrl}/${congressId}`);
  }

  createProposal(data: CreateProposal): Observable<Proposal> {
    return this.http.post<Proposal>(`${this.baseUrl}/proposal`, data);
  }

  getProposalById(id: number): Observable<Proposal> {
    return this.http.get<Proposal>(`${this.baseUrl}/proposal/${id}`);
  }

  getProposalByUserId(userId: number): Observable<Proposal> {
    return this.http.get<Proposal>(`${this.baseUrl}/proposal/user/${userId}`);
  }


  getProposalsByCongressId(congressId: number): Observable<Proposal[]> {
    return this.http.get<Proposal[]>(`${this.baseUrl}/congress/${congressId}`);
  }

  getProposalByCongressIdAndState(congressId: number,state:ProposalState): Observable<Proposal[]> {
    return this.http.get<Proposal[]>(`${this.baseUrl}/congress/${congressId}/${state}`);
  }

  updateProposal(id: number, data: ProposalState): Observable<Proposal> {
    const dataa = {state:data}
    return this.http.put<Proposal>(`${this.baseUrl}/proposal/${id}`, dataa);
  }


}

File: back/src/main/java/com/ayd2/congress/services/activity/ActivityService.java
package com.ayd2.congress.services.activity;

import java.util.List;

import com.ayd2.congress.dtos.acitivty.ActivityResponse;
import com.ayd2.congress.dtos.acitivty.NewActivityRequest;
import com.ayd2.congress.dtos.acitivty.NewProposalRequest;
import com.ayd2.congress.dtos.acitivty.ProposalResponse;
import com.ayd2.congress.dtos.acitivty.UpdateActivity;
import com.ayd2.congress.dtos.acitivty.UpdateProposal;
import com.ayd2.congress.exceptions.DuplicatedEntityException;
import com.ayd2.congress.exceptions.InvalidDateRangeException;
import com.ayd2.congress.exceptions.NotFoundException;
import com.ayd2.congress.models.Activities.ActivityEntity;
import com.ayd2.congress.models.Activities.ProposalEntity;
import com.ayd2.congress.models.Enums.ActivityType;
import com.ayd2.congress.models.Enums.ProposalState;

public interface ActivityService {
    ActivityResponse createActivity(NewActivityRequest request) throws NotFoundException,DuplicatedEntityException,InvalidDateRangeException;
    List<ActivityResponse> getActivitiesByCongressId(Long congressId) throws NotFoundException;
    List<ActivityResponse> getActivitiesByTypeAndCongressId(ActivityType type, Long congressId) throws NotFoundException;
    ProposalResponse createProposal(NewProposalRequest request) throws NotFoundException,DuplicatedEntityException;
    ProposalEntity getProposalById(Long id) throws NotFoundException;
    List<ProposalResponse> getProposalsByCongressId(Long congressId) throws NotFoundException;
    List<ProposalResponse> getProposalsByStateAndCongressId(ProposalState state, Long congressId) throws NotFoundException;
    List<ProposalResponse> getProposalByUserId(Long userId) throws NotFoundException;
    ActivityEntity getActivityById(Long id) throws NotFoundException;
    ProposalResponse updateProposal(Long id,UpdateProposal state) throws NotFoundException;
    ProposalResponse getProposalResponseById(Long id) throws NotFoundException;
    void deleteAcivity(Long activityId) throws NotFoundException;
    ActivityResponse updateActivity(Long id,UpdateActivity updateActivity) throws NotFoundException,DuplicatedEntityException,InvalidDateRangeException;
    
}
